lib/ex_doc/language/elixir.ex

```diff
@@ -0,0 +1,151 @@
+defmodule ExDoc.Language.Elixir do
+  @moduledoc false
+
+  @behaviour ExDoc.Language
+
+  @impl true
+  def filter_prefix_pattern(filter_prefix) do
+    if filter_prefix do
+      "Elixir.#{filter_prefix}*.beam"
+    else
+      "*.beam"
+    end
+  end
+
+  @impl true
+  def module_type(module) do
+    cond do
+      function_exported?(module, :__struct__, 0) and
+          match?(%{__exception__: true}, module.__struct__) ->
+        :exception
+
+      function_exported?(module, :__protocol__, 1) ->
+        :protocol
+
+      function_exported?(module, :__impl__, 1) ->
+        :impl
+
+      function_exported?(module, :behaviour_info, 1) ->
+        :behaviour
+
+      match?("Elixir.Mix.Tasks." <> _, Atom.to_string(module)) ->
+        :task
+
+      true ->
+        :module
+    end
```

Comment:
Shall we have a `:skip` and get rid of `skip_module?`?

Reply:
Or maybe make it return a tuple and the tuple says if it is shown or hidden by default?